packages/affiliates/libs/uniswap/utils.ts

```diff
@@ -32,3 +24,58 @@ export function convertValuesToString<T>(obj: { [k: string]: any }): T {
     })
   ) as T;
 }
+
+// This looks at a positions liquidity, and calculates the amount across a single tick based on full tick span
+export function liquidityPerTick(params: { liquidity: string; tickLower: string; tickUpper: string }) {
+  const { liquidity, tickLower, tickUpper } = params;
+  assert(BigInt(tickUpper) > BigInt(tickLower), "Upper tick must be > lower tick");
+  return (BigInt(liquidity) / (BigInt(tickUpper) - BigInt(tickLower))).toString();
+}
+
+// this is stronger than position, but compatible, since these fields have been validated to exist
+type ActivePosition = Position & {
+  liquidity: BigNumberish;
+  tickUpper: BigNumberish;
+  tickLower: BigNumberish;
+};
+
+// figure out if a position is active by looking a liquidity and combination of current tick and position bounds
+// this assume the upper tick is not inclusive, meaning if the upper tick matches the current tick, its not active
+export const IsPositionActive = (tick: BigNumberish) => (
+  position: Pick<Position, "liquidity" | "tickUpper" | "tickLower">
+): position is ActivePosition => {
+  assert(position.liquidity, "requires position liquidity");
+  assert(position.tickUpper, "requires position tickUpper");
+  assert(position.tickLower, "requires position tickLower");
+  if (BigInt(position.liquidity.toString()) === 0n) return false;
+  if (BigInt(tick.toString()) >= BigInt(position.tickUpper.toString())) return false;
+  if (BigInt(tick.toString()) < BigInt(position.tickLower.toString())) return false;
+  if (BigInt(position.tickUpper) <= BigInt(position.tickLower)) return false;
+  return true;
+};
+
+// percent: numerator, denominator and scale factor
+export function percent(val: string, sum: string, scale: string = (10n ** 18n).toString()) {
+  return ((BigInt(val) * BigInt(scale)) / BigInt(sum)).toString();
+}
+
+export function percentShares(
+  contributions: { [key: string]: string } = {},
+  sum?: string,
+  scale: string = (10n ** 18n).toString()
+) {
+  const defaultSum =
+    sum ||
+    Object.values(contributions)
+      .reduce((sum, val) => {
+        return sum + BigInt(val);
+      }, 0n)
+      .toString();
+
+  if (defaultSum == "0") return {};
```

Comment:
Even though this is typescript, we should probably still default to `===`. https://stackoverflow.com/a/57126110
```suggestion
  if (defaultSum === "0") return {};
```